src/org/opensolaris/opengrok/configuration/ConfigurationHelp.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2018, Chris Fraire <cfraire@me.com>.
+ */
+
+package org.opensolaris.opengrok.configuration;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import org.opensolaris.opengrok.authorization.AuthControlFlag;
+import org.opensolaris.opengrok.authorization.AuthorizationPlugin;
+import org.opensolaris.opengrok.authorization.AuthorizationStack;
+import org.opensolaris.opengrok.history.RepositoryInfo;
+import org.opensolaris.opengrok.index.Filter;
+import org.opensolaris.opengrok.index.IgnoredNames;
+
+/**
+ * Represents a utility class to present some user-readable help regarding
+ * {@link Configuration}.
+ */
+public class ConfigurationHelp {
+    /**
+     * Gets sample content for a configuration XML file.
+     * @return a defined instance
+     * @throws RuntimeException if an error occurs producing the sample
+     */
+    public static String getSamples()
+            throws RuntimeException {
+
+        Configuration conf = new Configuration();
+        Class klass = conf.getClass();
+        klass.getDeclaredMethods();
```

Comment:
This seems to throw away the returned array of Method objects. Is this necessary ?

Reply:
OK deleted